src/core/thread/mle_router.cpp

```diff
@@ -2540,7 +2540,7 @@ void MleRouter::HandleChildUpdateRequest(const Message &         aMessage,
         {
             child->SetCslChannel(static_cast<uint8_t>(cslChannel.GetChannel()));
         }
-        else
+        else if (child->GetCslChannel() == 0)
```

Comment:
According to the 802.15.4-2015 spec, channel 0 is a valid channel. The frequency of the channel 0 is 868.3MHz. If someone implements the channel 0 in the future, there will be a problem here.

Reply:
Here `child->GetCslChannel() == 0` means it's unset. I think we define a invalid value instead of 0 to replace it later.

Reply:
It sounds reasonable.